Geometry_Engine/Compute/GrahamScan.cs

```diff
@@ -0,0 +1,151 @@
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Quantities.Attributes;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Implements the GrahamScan algorithm to determine the convex hull of a list of points contained within the XY Plane.")]
+        [Input("pts", "The points to determine the convex hull contained within the XY Plane.")]
+        [Input("tolerance", "Geometrical tolerance to be used in the method.", typeof(Length))]
+        [Output("c", "The convex hull of the point list, no repeat points are returned.")]
+        public static List<Point> GrahamScan(List<Point> pts, double tolerance = Tolerance.MicroDistance)
+        {
+            if (pts.IsNullOrEmpty())
+            {
+                Base.Compute.RecordError("The point list is null or empty.");
+                return pts;
+            }
+            if (pts.Count < 3)
+            {
+                Base.Compute.RecordError("The point list must be greater than three to determine the Convex Hull.");
+                return pts;
+            }
+
+            // Check that the points are all wihtin the same plane 
+            Plane plane = pts.FitPlane(tolerance);
+            if (!Query.IsInPlane(pts, Plane.XY))
+            {
+                Base.Compute.RecordError("The point list needs to be within the XY Plane.");
```

Comment:
That was a hangover when I tried (unsuccessfully) to make it orientation agnostic. Have opted for Option 3 because the `Plane` is needed to get the local axis to `Orient()` them.

Reply:
I think it could be done if you take normal of the plane fit and create a `Cartesian` coordinate system  with arbitrary X and Y, to then use it to orient to and back from the global. Wouldn't that work?